src/katcp_fpga.py

```diff
@@ -537,4 +537,19 @@ def unhandled_inform(self, msg):
         if self.unhandled_inform_handler is not None:
             self.unhandled_inform_handler(msg)
 
+    def check_phy_counter(self):
+        request_args = ((0, 0), (0, 1), (1, 0), (1, 1))
+        for arg in request_args:
+            try:
+                result0 = self.katcprequest('phywatch', request_args=arg)
+                time.sleep(1)
+                result1 = self.katcprequest('phywatch', request_args=arg)
+                assert result0[0].reply_ok()  # useful?
+                assert (int(result1[0].arguments[1].replace('0x', ''), base=16) -
+                        int(result0[0].arguments[1].replace('0x', ''), base=16)) != 0
+            except:
```

Comment:
Not a good idea to do a 'naked' except, since bugs and other problems will be hidden. You should catch a specific exception, so probably 'except AssertError'.

But I'm not sure if you need to use the whole try-except with assert statements at all. Just use normal conditional logic, e.g.: 

```
if  (int(result1[0].arguments[1].replace('0x', ''), base=16) -
     int(result0[0].arguments[1].replace('0x', ''), base=16))  == 0:
   #log stuff
   return False
```